lib/good_job/active_job_extensions/concurrency.rb

```diff
@@ -41,9 +41,9 @@ module Concurrency
           next if key.blank?
 
           GoodJob::Job.new.with_advisory_lock(key: key, function: "pg_advisory_lock") do
-            perform_concurrency = GoodJob::Job.unscoped.where(concurrency_key: key).advisory_locked.count
+            allowed_jobs = GoodJob::Job.unscoped.where(concurrency_key: key).advisory_locked.order(performed_at: :asc).limit(limit).collect(&:active_job_id)
```

Comment:
This is genius! I have some functional and stylistic suggestions:

```suggestion
            allowed_active_job_ids = GoodJob::Job.unscoped.where(concurrency_key: key).advisory_locked.order(Arel.sql "COALESCE(performed_at, scheduled_at, created_at) ASC").limit(limit).pluck(&:active_job_id)
```

- `performed_at` is only present if `GoodJob.preserve_job_records` is true so I want to suggest the COALESCE to ensure that there is always a deterministic value and not null/nil: https://github.com/bensheldon/good_job/blob/9c2b9f74396881a2f22b5a38681aefabe4405e40/lib/good_job/job.rb#L259-L260
- using ActiveRecord's `pluck` for a more efficient query
- being longwinded in variable names to be clear if a variable holds a good_job_id or an active_job_id

Reply:
Awesome, thank you!

I staunchly support overly descriptive variable names—and not thinking about `performed_at` being empty is one of those unknown unknowns I knew I was missing!

I didn’t go with `pluck` because I noticed there’s an accessor method for `Job#active_job_id` that falls back to `serialized_params['job_id']`. I’m guessing that might be for unsaved records, which isn’t relevant here, and so `pluck` is fine? Just want to double-check!

Reply:
There is a column for `active_job_id`... but it's not guaranteed to exist for a user if they haven't kept up with GoodJob migrations. I am comfortable leaving it this situation where the column may not exist and raise a database exception (SEMVER is a lot of work to get perfect).

But, if you wanted to go the extra mile, you could do a feature check and implement it both the more performant and less performant ways with a deprecation notice like this: https://github.com/bensheldon/good_job/blob/811fb264bd29626331caaa9609bf17151e9ca005/lib/good_job/job.rb#L210-L223

I think this could be a fairly hot code path, so the better query is important.